src/python/WMCore/MicroService/MSOutput/MSOutput.py

```diff
@@ -594,7 +596,9 @@ def msOutputProducer(self, requestRecords):
                 msg = "%s General Error from pipeline. Err: %s. " % (pipeLineName, str(ex))
                 msg += "Giving up Now."
                 self.logger.exception(str(ex))
-                break
+                workflowname = request.get("_id", "none")
+                self.alertGenericError("producer", workflowname, msg, str(ex), str(request))
```

Comment:
Instead of writing "consumer" here, perhaps pass the variable `self.mode`

Reply:
thanks!